server/src/main/java/io/deephaven/server/barrage/BarrageMessageProducer.java

```diff
@@ -2133,17 +2219,23 @@ BarrageMessage getSnapshot(
             final BitSet columnsToSnapshot,
             final RowSet positionsToSnapshot,
             final RowSet reversePositionsToSnapshot) {
-        if (onGetSnapshot != null) {
+        if (onGetSnapshot != null && onGetSnapshotIsPreSnap) {
             onGetSnapshot.run();
         }
 
         // TODO: Use *this* as snapshot tick source for fail fast.
         // TODO: Let notification-indifferent use cases skip notification test
```

Comment:
nit, these are missing issues

Reply:
Looks like Ryan added them; I'll keep up with him to figure something out.